### PR TITLE
Additional app-fixes [artv3/appfixes]

### PR DESCRIPTION
**Pull request by artv3**
This PR fixes minor things in MFEM needed in applications running on GPUs.

### CODE REVIEW

File: fem/pgridfunc.cpp [hunk 1]
```diff
@@ -225,10 +225,9 @@ void ParGridFunction::ExchangeFaceNbrData()
    MPI_Request *recv_requests = requests + num_face_nbrs;
    MPI_Status  *statuses = new MPI_Status[num_face_nbrs];
 
-   const double *h_data = this->HostRead();
    for (int i = 0; i < send_data.Size(); i++)
    {
-      send_data[i] = h_data[send_ldof[i]];
+      send_data[i] = data[send_ldof[i]];
```
Comment (v-dobrev) [inline fem/pgridfunc.cpp]: I think this change is wrong -- we want this method to work when `data` is in GPU memory too.

What is your argument for the change?

File: linalg/vector.hpp [hunk 2]
```diff
@@ -141,11 +141,11 @@ class Vector
    inline void NewMemoryAndSize(const Memory<double> &mem, int s, bool own_mem);
 
    /// Reset the Vector to be a reference to a sub-vector of @a base.
-   inline void MakeRef(Vector &base, int offset, int size);
+   inline void MakeRef(const Vector &base, int offset, int size);
```
Comment (v-dobrev) [inline linalg/vector.hpp]: I don't think adding `const` here is the correct way to create sub-vectors of a `const Vector`. We do not want the interface to remove `const`-ness implicitly. This should be done explicitly in user code with `const_cast<Vector &>(...)`.

File: fem/pgridfunc.cpp (hunk 1 above)
Comment (artv3) [reply fem/pgridfunc.cpp]: Sorry about this.... I mixed up files. You are right, the update should be with exchanging face nbr data a few lines below. I've just pushed the correct fix.

File: fem/pgridfunc.cpp [hunk 3]
```diff
@@ -240,7 +241,7 @@ void ParGridFunction::ExchangeFaceNbrData()
                 send_offset[fn+1] - send_offset[fn],
                 MPI_DOUBLE, nbr_rank, tag, MyComm, &send_requests[fn]);
 
-      MPI_Irecv(&face_nbr_data(recv_offset[fn]),
+      MPI_Irecv(&h_face_nbr_data[recv_offset[fn]],
```
Comment (artv3) [inline fem/pgridfunc.cpp]: @v-dobrev , this is the change I meant to push.

File: linalg/vector.hpp (hunk 2 above)
Comment (artv3) [reply linalg/vector.hpp]: Okay, we can make that change in our application code.

Comment (artv3) [reply linalg/vector.hpp]: Just reverted the change.

**Comment by v-dobrev** [comment]
Merging directly since the target branch is not master.

**Pull request merged**